x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -91,18 +91,15 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 	// Get startTime and endTime
-	startTime, endTime, err := aws.GetStartTimeEndTime(m.DurationString)
-	if err != nil {
-		return errors.Wrap(err, "error GetStartTimeEndTime")
-	}
+	startTime, endTime := aws.GetStartTimeEndTime(m.Period)
 
 	// Get listMetrics and namespacesTotal from configuration
 	listMetrics, namespacesTotal := readCloudwatchConfig(m.CloudwatchConfigs)
 	for _, regionName := range m.MetricSet.RegionsList {
 		awsConfig := m.MetricSet.AwsConfig.Copy()
 		awsConfig.Region = regionName
 		svcCloudwatch := cloudwatch.New(awsConfig)
-		err := createEvents(svcCloudwatch, listMetrics, regionName, m.PeriodInSec, startTime, endTime, report)
+		err := createEvents(svcCloudwatch, listMetrics, regionName, int(m.Period.Seconds()), startTime, endTime, report)
```

Comment:
This could be passed as `time.Duration` and converted to seconds only when needed (in this case when creating the `cloudwatch.MetricDataQuery`).

Reply:
Good point, totally missed it! Will make the change, thanks!